src/core/utils/Accumulator.hpp

```diff
@@ -0,0 +1,111 @@
+#ifndef CORE_UTILS_ACCUMULATOR
+#define CORE_UTILS_ACCUMULATOR
+
+#include <boost/archive/binary_iarchive.hpp>
+#include <boost/archive/binary_oarchive.hpp>
+#include <boost/serialization/string.hpp>
+#include <boost/serialization/vector.hpp>
+#include <boost/iostreams/device/array.hpp>
+#include <boost/iostreams/stream.hpp>
+
+namespace Utils {
+
+template <typename T> struct AccumulatorData {
+  AccumulatorData()
+      : m_mean_old(0.0), m_mean_new(0.0), m_variance_old(0.0),
+        m_variance_new(0.0) {}
+  T m_mean_old;
```

Comment:
If think you only need to store one mean and one variance, see below. Also don't call the members `m_` if they are not private.